modin/pandas/utils.py

```diff
@@ -104,6 +104,29 @@ def remote(self, *args):
         return result
 
 
+def post_task_gc(func):
+    """Perform garbage collection after the task is executed.
+
+    Usage:
+        ```
+        @ray.remote
+        @post_task_gc
+        def memory_hungry_op():
+            ...
+        ```
+    Note:
+        - This will invoke the GC for the entire process. Expect
+          About 100ms latency. 
```

Comment:
Is the latency still 100ms for gc'ing groupby on a smaller dataframe? Might be worth thinking about the speed-memory tradeoff.

Reply:
Can we potentially be more intelligent about when to run rc? Perhaps time the function run and as long as it is 5x the expected gc runtime, we gc? This way we cap speed performance regression to 20%.

Reply:
Yes we should be more intelligent. I can add a basic heuristic for now, per your suggestion.  

However, considering the use case, we are not faster than pandas on small dataframe. 

The memory issue can be a problem for us already if the dataframe is few hundred MBs. For now, this is really about just making sure we can even do groupby without crashing (due to OOM).